src/test/compile-fail/issue-25385.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+macro_rules! foo {
+    ($e:expr) => { $e.foo() }
+    //~^ ERROR no method named `foo` found for type `i32` in the current scope
+}
+
+fn main() {
+    let a = 1i32;
+    foo!(a);
```

Comment:
As it is now, the invocation site is displayed with a note, e.g. for the test case the relevant output would be:

    src/test/compile-fail/issue-25385.rs:13:23: 13:28 error: no method named `foo` found for type `i32` in the current scope
    src/test/compile-fail/issue-25385.rs:13     ($e:expr) => { $e.foo() }
    src/test/compile-fail/issue-25385.rs:19:5: 19:13 note: in this expansion of foo! (defined in src/test/compile-fail/issue-25385.rs)

So it includes a backtrace to the invocation site. But in order to find out which parameter `$e` is expanded to one would have to look at the macro definition.